File: Ejercicios/Ejercicio 1/Ejercicio 1-Java/Sayajin.java
import java.util.Arrays;

public class Sayajin{
	private int ki;
	private int base;
	private String nombre;
	private int fase;

	public Sayajin(){
		this(0,0,"",0);
	}

	public Sayajin(int ki,int base,String nombre,int fase){
		this.ki = ki;
		this.base = base;
		this.nombre = nombre;
		this.fase = fase;
	}

	public String getNombre(){ return nombre; }
	public void setNombre(String nombre){ this.nombre = nombre; }

	public int getKi(){ return ki; }
	public void setKi(int ki){ this.ki = ki; }

	public int getBase(){ return base; }
	public void setBase(int base){ this.base = base; } 

	public int getFase(){ return fase; }
	public void setFase(int fase){ this.fase = fase; }

	@Override
	public String toString(){
		return nombre+"......Fase-"+fase+"\tKi-"+ki+"\t\tNivel base de poder:"+base;
	}

	public void visor(){
		System.out.println("El poder de pelea de tu Sayajin es......." + ki);
	}

	public void transformar(int fase){
		this.fase = fase;

		if(fase == 0){ ki = base; }
		if(fase == 1){ ki = base * 50; }
		if(fase == 2){ ki = base * 100; }
		if(fase == 3){ ki = base * 400; }
		if(fase == 4){ ki = base * 4000; }
	}

	public Sayajin fusionar(Sayajin s){
		Sayajin nvo = new Sayajin();

		nvo.setFase( fase + s.getFase() );
		nvo.setBase( base + s.getBase() );
		nvo.setKi( ki + s.getKi() );
		nvo.setNombre( nombre.substring(0,(nombre.length()/2)) + s.getNombre().substring( (s.getNombre().length()/2),(s.getNombre().length()) ) );

		return nvo;
	}
}

File: Ejercicios/Ejercicio 1/Ejercicio 1-C/Sayajin.c
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>
#include <string.h>
#include "Sayajin.h"

sayajin createSayajin(){
	sayajin s;
	int i;

	s = (sayajin)malloc(sizeof(Sayajin)*3);

	if(s=NULL){
		printf("Memoria insuficiente\n");
		exit(1);
	}

	for(i = 0; i < 3; ++i){
		s[i].nombre = (char*)malloc(sizeof(char));
		s[i].ki = 0;
		s[i].base = 0;
		s[i].fase = 0;
	}

	return s;
}

//	char* nombre;
//	unsigned long ki;
//	unsigned long base;
//	int fase;

sayajin setSayajin(sayajin s,int id){
	printf("Generando sayajines....\n");
	switch(id){
		case 1: strcpy(s[id-1].nombre,"Goku");
				s[id-1].ki = 5000;
				s[id-1].base = 5000;
				s[id-1].fase = 0;
		break;
		case 2: strcpy(s[id-1].nombre,"Vegeta");
				s[id-1].ki = 4800;
				s[id-1].base = 4800;
				s[id-1].fase = 0;
		break;
		case 3: strcpy(s[id-1].nombre,"Trunks");
				s[id-1].ki = 4000;
				s[id-1].base = 4000;
				s[id-1].fase = 0;
		break;
	}

	return s;
}

void visor(sayajin s){
	printf("\nEl poder de pelea de %s es de... %lu\n",s->nombre,s->ki);
}

void transformar(sayajin s,int fase){
	s->fase = fase;

	if(fase == 0){ s->ki = s->base; }
	if(fase == 1){ s->ki = s->base * 50; }
	if(fase == 2){ s->ki = s->base * 100; }
	if(fase == 3){ s->ki = s->base * 400; }
	if(fase == 4){ s->ki = s->base * 4000; }
}

sayajin fusionar(sayajin s,sayajin ss){
	sayajin nvo;

	return nvo;
}